persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_51_gsheet/generated/entities/persist_client.bal

```diff
@@ -0,0 +1,497 @@
+        record {}[] outputArray = check from record {} 'object in workspacesStream
+            outer join var building in buildingsStream on ['object.locationBuildingCode] equals [building?.buildingCode]
+            select persist:filterRecord({
+                ...'object,
+                "building": building
+            }, fields);
+        return outputArray.toStream();
+    }
+
+    private function queryOneWorkspaces(anydata key) returns record {}|error {
+        stream<Workspace, error?> workspacesStream = self.queryWorkspacesStream();
+        stream<Building, error?> buildingsStream = self.queryBuildingsStream();
+        error? unionResult = from record {} 'object in workspacesStream
+            outer join var building in buildingsStream on ['object.locationBuildingCode] equals [building?.buildingCode]
+            do {
+                return {
+                    ...'object,
+                    "building": building
+                };
+            };
+        if unionResult is error {
+            return <persist:Error>error(unionResult.message());
+        }
+        return <persist:InvalidKeyError>error("Invalid key: " + key.toString());
+    }
+
+    private isolated function queryWorkspacesStream(WorkspaceTargetType targetType = <>) returns stream<targetType, error?> = @java:Method {
+        'class: "io.ballerina.stdlib.persist.datastore.GoogleSheetsProcessor",
+        name: "queryStream"
+    } external;
+
+    isolated resource function get buildings(BuildingTargetType targetType = <>) returns stream<targetType, persist:Error?> = @java:Method {
+        'class: "io.ballerina.stdlib.persist.datastore.GoogleSheetsProcessor",
+        name: "query"
+    } external;
+
+    isolated resource function get buildings/[string buildingCode](BuildingTargetType targetType = <>) returns targetType|persist:Error = @java:Method {
+        'class: "io.ballerina.stdlib.persist.datastore.GoogleSheetsProcessor",
+        name: "queryOne"
+    } external;
+
+    isolated resource function post buildings(BuildingInsert[] data) returns string[]|persist:Error {
+        _ = check self.persistClients.get(BUILDING).runBatchInsertQuery(data);
+        return from BuildingInsert inserted in data
+            select inserted.buildingCode;
+    }
+
+    isolated resource function put buildings/[string buildingCode](BuildingUpdate value) returns Building|persist:Error {
+        _ = check self.persistClients.get(BUILDING).runUpdateQuery(buildingCode, value);
+        return self->/buildings/[buildingCode].get();
+    }
+
+    isolated resource function delete buildings/[string buildingCode]() returns Building|persist:Error {
+        Building result = check self->/buildings/[buildingCode].get();
+        _ = check self.persistClients.get(BUILDING).runDeleteQuery(buildingCode);
+        return result;
+    }
+
+    private function queryBuildings(string[] fields) returns stream<record {}, persist:Error?>|error {
```

Comment:
This was changed to specific errors, https://github.com/ballerina-platform/module-ballerina-persist/blob/e459c7b8cc8656c229e52bb233c0d6f97fbb8d0e/ballerina/tests/googlesheets_rainier_generated_client.bal#L370